### DISCUSSION TITLE
source files in cache dir.

### DISCUSSION THREAD

**Discussion by gnaga** [SCons/scons#4061] [Q&A]
Hi Team,

when cache is enabled, the source files (c++ and c files) are also copied to cachedir. Can we copy and get only generated files from cons instead of source code.

ex: 
Retrieved `export/src/test.c++' from cache
Retrieved `export/src/test1.c++' from cache
Retrieved `build/test.o' from cache
Retrieved `build/test1.o' from cache

Thanks in advance.

**Comment by bdbaddog** [comment]
Is export/src/test.c++ a generated file?

**Reply by gnaga** [reply]
yes it is. But can we restrict no to copying or retrieving ASCII files? There will be many small generated files that will put more load of copy if the cache is put in NFS.

**Reply by bdbaddog** [reply]
You can't list the files to not be cached by some suffix wildcard.
You can use:
https://scons.org/doc/production/HTML/scons-man.html#f-NoCache

on a target by target basis.
You could use SCons' `Glob()` function to glob through your dirs to find `*.c`'s and set `NoCache()` on those.

Buy why skip caching any particular generated file, in general it's only going to be faster.

**Reply by gnaga** [reply]
If the cache is in NFS (Shared cache), and thousands of small files, copying them would take time. If build machines have more vcpu (24+), generating will be faster. Also if the shared cache is used by 10's of users then NFS r/w will be slow. 
So If we have this option, then builds might be faster.

Also, I will check the option of Glob() and NoCache().

**Reply by bdbaddog** [reply]
As I've stated, there's ways for you to exclude files from cache already.
Adding a wildcard based on file extensions hasn't been requested thus far, and cachedir is in wide use on some fairly large codebases (3m+ LOC and more).

**Comment by bdbaddog** [comment]
Closing this issue.
If you'd like to submit a PR with this functionality it might be considered.
